central/auth/m2m/verifier.go

```diff
@@ -48,7 +48,10 @@ func (a authenticatedRoundTripper) RoundTrip(req *http.Request) (*http.Response,
 	// First try without any auth header
 	resp, err := a.roundTripper.RoundTrip(req)
 	if resp.StatusCode >= 400 {
```

Comment:
Couldn't `resp` be `nil` on error?

Reply:
Yep, not sure how I missed this. Thanks for catching it!  Should be fixed in 607117ed0d407bfe3097d5658d55b3eb52cb2b56